### PR TITLE
Mention dot-prefixed patterns in hidden index docs

### PR DESCRIPTION
**Pull request by AthenaEryma**
The hidden index docs did not mention that dot-prefixed patterns
default to matching hidden indices. This PR adds a note explaining
the behavior and why it's like that.

### CODE REVIEW

File: docs/reference/api-conventions.asciidoc [hunk 1]
```diff
@@ -177,6 +177,11 @@ For most APIs, wildcard expressions do not match hidden data streams and indices
 by default. To match hidden data streams and indices using a wildcard
 expression, you must specify the `expand_wildcards` query parameter.
 
+Alternatively, querying an index pattern starting with a dot, such as
+`.watcher_hist*`, will match hidden indices by default. This is intended to
+Unix file-globbing behavior and provide a smoother transition path to
```
Comment (not-napoleon) [inline docs/reference/api-conventions.asciidoc:182]:
```suggestion
mirror Unix file-globbing behavior and provide a smoother transition path to
```

**Review by not-napoleon — APPROVED** [review]
I think you missed a word.  Otherwise looks good, thanks for jumping on it.

**Review by jrodewig — APPROVED** [review]
LGTM with @not-napoleon's suggestion.

**Comment by gwbrown** [comment]
The build failure was complaining about cross-document links in Kibana's docs, I don't think it has anything to do with these changes.

@elasticmachine run elasticsearch-ci/docs

**Pull request merged**